src/nupic/algorithms/Cells4.cpp

```diff
@@ -1036,11 +1029,12 @@ void Cells4::updateInferenceState(const std::vector<UInt> & activeColumns)
   _infActiveStateT1 = _infActiveStateT;
   _infPredictedStateT1 = _infPredictedStateT;
   memcpy(_cellConfidenceT1, _cellConfidenceT, _nCells * sizeof(_cellConfidenceT[0]));
+  memset(_cellConfidenceT, 0, _nCells * sizeof(_cellConfidenceT[0]));
```

Comment:
Oh I see the comment on the line later that zeros the values. Now there are three different spots that memset `_cellConfidenceT` which seems excessive. Ideally there would just be one spot as before.